OMEdit/OMEditGUI/Editors/MetaModelEditor.cpp

```diff
@@ -60,6 +60,7 @@ MetaModelEditor::MetaModelEditor(QWidget *pParent)
   : BaseEditor(pParent), mLastValidText(""), mTextChanged(false), mForceSetPlainText(false)
 {
   mXmlDocument = XMLDocument(this);
+  connect(mpPlainTextEdit,SIGNAL(textChanged()),this,SLOT(updateAllOrientations()));
```

Comment:
We shouldn't do this. `textChanged` signal is emitted whenever `setPlainText` is called. We use `setPlainText` a lot in the code which means numerous calls to `updateAllOrientations`.

I believe you added this so if user modifies something in the code that should reflect in the 3d viewer. In my opinion we shouldn't update the 3d viewer when user is editing something in the text view because the code might be incomplete. Only update if user press save or switch back to diagram view. See `metaModelEditorTextChanged` which is called if user has modified something and is switching back to diagram view.